examples/ingressRaw.dhall

```diff
@@ -1,81 +1,104 @@
-in let IntOrString    = ../default/io.k8s.apimachinery.pkg.util.intstr.IntOrString.dhall
+in  let TLS = ./../types/io.k8s.api.extensions.v1beta1.IngressTLS.dhall
 
--- Our Service type
-in let Service = ./Config.dhall
-in let Config = { services : List Service }
+in  let Rule = ./../types/io.k8s.api.extensions.v1beta1.IngressRule.dhall
 
--- A function to generate an ingress given a configuration
-in let mkIngress : Config -> Ingress =
+in  let RuleVal =
+          ./../types/io.k8s.api.extensions.v1beta1.HTTPIngressRuleValue.dhall
 
-  \(config : Config) ->
+in  let Spec = ./../types/io.k8s.api.extensions.v1beta1.IngressSpec.dhall
 
-  -- Given a service, make a TLS definition with their host and certificate
-     let makeTLS = \(service : Service) ->
-    { hosts = Some (List Text) [ service.host ]
-    , secretName = Some Text "${service.name}-certificate"
-    }
+in  let Ingress = ./../types/io.k8s.api.extensions.v1beta1.Ingress.dhall
 
-  -- Given a service, make an Ingress Rule
-  in let makeRule = \(service : Service) ->
-    { host = Some Text service.host
-    , http = Some RuleVal
-      { paths = [ { backend = { serviceName = service.name
-                              , servicePort = IntOrString.Int 80
-                              }
-                  , path = None Text
-                  }
-                ]
-      }
-    }
-
-  -- Nginx ingress requires a default service as a catchall
-  in let defaultService =
-    { name = "default"
-    , host = "default.example.com"
-    , version = " 1.0"
-    }
+in  let defaultIngress =
+          ./../default/io.k8s.api.extensions.v1beta1.Ingress.dhall
 
-  -- List of services
-  in let services = config.services # [ defaultService ]
+in  let defaultMeta =
+          ./../default/io.k8s.apimachinery.pkg.apis.meta.v1.ObjectMeta.dhall
 
-  -- Some metadata annotations
-  -- NOTE: `dhall-to-yaml` will generate a record with arbitrary keys from a list
-  -- of records where mapKey is the key and mapValue is the value of that key
-  in let genericRecord = List { mapKey : Text, mapValue : Text }
-  in let kv = \(k : Text) -> \(v : Text) -> { mapKey = k, mapValue = v }
+in  let defaultSpec =
+          ./../default/io.k8s.api.extensions.v1beta1.IngressSpec.dhall
 
-  in let annotations = Some genericRecord
-    [ kv "kubernetes.io/ingress.class"      "nginx"
-    , kv "kubernetes.io/ingress.allow-http" "false"
-    ]
+in  let IntOrString =
+          ./../default/io.k8s.apimachinery.pkg.util.intstr.IntOrString.dhall
 
-  -- Generate spec from services
-  in let spec = defaultSpec //
-    { tls   = Some (List TLS)  (map Service TLS  makeTLS  services)
-    , rules = Some (List Rule) (map Service Rule makeRule services)
-    }
+in  let Service = ./Config.dhall
 
-  in defaultIngress
-    { metadata = defaultMeta
-      { name = "nginx" } //
-      { annotations = annotations }
-    } //
-    { spec = Some Spec spec }
+in  let Config = { services : List Service }
 
+in  let mkIngress
```

Comment:
Nice refactoring. You run the file through `dhall format` right?
If yes, note that it strips out the comments that are not at the top of the file. I'd like to keep a good amount of comments in these example files so that a reader knows what's going on

Reply:
Eh, I undid it again. Apparently I forgot to undo it on the other file.